span_test.go

```diff
@@ -118,6 +118,60 @@ func TestSpanParentID(t *testing.T) {
 	assert.Equal(t, model.SpanID(parentID), payloads.Spans[0].ParentID)
 }
 
+func TestSpanLink(t *testing.T) {
```

Comment:
I think this test could be simplified, so it just asserts bits related to span links. Assertions on parent ID being non-empty and such shouldn't be necessary; that is covered by other tests.

Suggestions:
 - create a single span, like `tracer.StartSpan("name", "type", apm.SpanID{}, apm.SpanOptions{Links: ...})`
 - hand craft the span links, like `apm.SpanLink{Trace: apm.TraceID{...}, Span: apm.SpanID{...}}`

You should end up with something like:

```go
links := []apm.SpanLink{...} // mock IDs
span := tracer.StartSpan("name", "type", apm.SpanID{}, apm.SpanOptions{Links: links})
defer span.End()

tracer.Flush(nil)
payloads := tracer.Payloads()
require.Len(t, payloads.Spans, 1)

expectedLinks := []model.SpanLink{...} // also mock IDs, same values as in links
assert.Equal(t, expectedLinks, payloads.Spans[0].Links)
```

Reply:
Thanks for the tip! :bow: 

I've pushed some changes. I still had to keep the loop as `apm.SpanLink` and `model.SpanLink` are two different types so the equality check would fail.